### PR TITLE
Landscape selection

### PR DESCRIPTION
**Pull request by DamnClin**
Close https://github.com/jhipster/jhipster-lite/issues/3050
Close https://github.com/jhipster/jhipster-lite/issues/3031

### CODE REVIEW

File: src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    return this.memoizedDependantModules.get(module, () =>
+      this.allDependantSelectedModules(module, element => this.elementDependantModules(element))
+    );
+  }
+
+  private elementDependantModules(element: ElementSlug): ElementSlug[] {
+    const directDependantModules = this.findElement(element).dependantModules;
+
+    return this.moduleFeature(element)
+      .map(feature => {
```
Comment (hdurix) [inline src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts:102]: maybe extract this in a function with proper name

File: src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        if (this.switchModuleInFeature(feature, element)) {
+          return directDependantModules;
+        }
+
+        return [...directDependantModules, ...feature.dependantModules];
+      })
+      .orElse(directDependantModules);
+  }
+
+  public modulesToDeselect(module: ElementSlug): ElementSlug[] {
```
Comment (hdurix) [inline src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts:112]: modulesToUnselect

File: src/main/webapp/app/module/primary/landscape/Landscape.component.ts [hunk 3]
```diff
@@ -112,25 +100,52 @@ export default defineComponent({
...
+        return false;
       }
 
-      const isDependentModule = landscape.value.value().getModule(emphasizedModule.value).hasDependency(module);
-      if (isDependentModule) {
-        return flavoredClass('-highlighted');
+      return emphasizedModule.value === module || landscapeValue().moduleSelectionTree(emphasizedModule.value).includes(module);
+    };
+
+    const deselectionHighlightClass = (module: string): string => {
```
Comment (hdurix) [inline src/main/webapp/app/module/primary/landscape/Landscape.component.ts:127]: unselectionHighlightClass

File: src/main/webapp/app/module/primary/landscape/Landscape.component.ts [hunk 4]
```diff
@@ -150,19 +166,87 @@ export default defineComponent({
...
       return '';
     };
 
+    const emphasizedDependency = (startingElement: string): boolean => {
+      if (emphasizedModule.value === undefined) {
+        return false;
+      }
+
+      const currentElementEmphasized = emphasizedModule.value === startingElement;
+      const dependencyEmphasied = landscapeValue().moduleSelectionTree(emphasizedModule.value).includes(startingElement);
```
Comment (hdurix) [inline src/main/webapp/app/module/primary/landscape/Landscape.component.ts:194]: dependencyEmphasized

File: src/test/javascript/spec/module/primary/landscape/Landscape.spec.ts [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+    it('Should not allow selection creating a bad switch in feature', async () => {
+      const wrapper = await componentWithLandscape();
+
+      await clickModule('maven', wrapper);
+      await clickModule('gitlab-maven', wrapper);
+
+      expect(wrapper.find(wrappedElement('gitlab-gradle-module')).classes()).toContain('-not-selectable');
+    });
+  });
+  //github.com/jhipster/jhipster-lite/issues/3129
```
Comment (hdurix) [inline src/test/javascript/spec/module/primary/landscape/Landscape.spec.ts:320]: is this comment still wanted?

File: src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts (hunk 1 above)
Comment (DamnClin) [reply src/main/webapp/app/module/primary/landscape/ComponentLandscape.ts:102]: Not a good idea with the shared const

**Review by hdurix — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Adding a bounty on this one, to complete the landscape page, which is simply awesome
Plz claim it, @DamnClin

**Comment by DamnClin** [comment]
Claimed https://opencollective.com/generator-jhipster/expenses/92092

**Comment by pascalgrimaud** [comment]
@DamnClin : approved

**Pull request merged**